x-pack/plugins/canvas/public/components/workpad_page/event_handlers.js

```diff
@@ -44,37 +44,34 @@ const setupHandler = (commit, canvasOrigin) => {
 const handleMouseMove = (
   commit,
   { clientX, clientY, altKey, metaKey, shiftKey, ctrlKey },
-  isEditable,
   canvasOrigin
 ) => {
-  if (isEditable) {
-    const { x, y } = localMousePosition(canvasOrigin, clientX, clientY);
-    commit('cursorPosition', { x, y, altKey, metaKey, shiftKey, ctrlKey });
-  }
+  const { x, y } = localMousePosition(canvasOrigin, clientX, clientY);
+  commit('cursorPosition', { x, y, altKey, metaKey, shiftKey, ctrlKey });
 };
 
 const handleMouseLeave = (commit, { buttons }) => {
-  if (buttons !== 1) {
+  if (buttons !== 1 && commit) {
```

Comment:
Q: None of the other handlers have this check for `commit`, curious why you added it here.

Reply:
Good question! With the other handlers, the `commit` is always there. But `mouseLeave` - therefore its handler - can get triggered even if we're no longer on that page - maybe as the DOM subtree itself changes into something specced by `SimplePage` -, and an error arose. I'm not 100% sure if it follows from the `mouseLeave` etc. related web standards or a browser frailty similar to https://bugs.chromium.org/p/chromium/issues/detail?id=515921 and https://stackoverflow.com/questions/37163602/how-to-stop-firefox-firing-mouseleave-when-dom-node-is-moved-in-the-dom but wanted to ensure we don't call something that might no longer be there.

Reply:
Ah, makes sense. Would it be worth adding to the other handlers too, even if only for consistency?

Reply:
Sure thing, will do!

Reply:
Done!